### ISSUE TITLE
#followers should be a Relation, not array

### ISSUE THREAD

**Issue by rstacruz** [tcocca/acts_as_follower#58]
The `followers` method uses .to_a:

```rb
      def followers(options={})
        followers_scope = followers_scoped.unblocked
        followers_scope = apply_options_to_scope(followers_scope, options)
        followers_scope.to_a.collect{|f| f.follower}
      end
```

It'd be better if this returned a relation so I can do, say, `post.followers.to_sql` or `post.followers.select(:id)`.

**Comment by andrewhamon** [comment]
@rstacruz because of the polymorphic nature of the `follows` table, that can't really be done, since followers could contain any number of types.

If your app only has one type that acts as a follower, you can use `followers_by_type('Type')`, where `'Type'` is the follower type of interest. This returns an active record relation (at least in the latest released version) despite the docs saying it will return an array.

**Comment by rstacruz** [comment]
that's awesome! thats definitely what i'm looking for.

**Comment by andrewhamon** [comment]
Sweet! If that solves your issue then it would be ideal if you closed it, since I don't think the maintainers are going to any time soon.

**Comment by rstacruz** [comment]
would be nice to have it documented, though.
